### ISSUE TITLE
Making the parent-child relationship fieldname configurable

### ISSUE THREAD

**Issue by tdido** [agshumate/Liftoff#78]
Hello.

Some GFF3 files use the "Derives_from" field instead of the "Parent" field to establish a parent-child relationship (see [an example from miRBase](ftp://mirbase.org/pub/mirbase/CURRENT/genomes/ath.gff3).

Would you consider a pull request to add an argument that makes this configurable?

**Comment by agshumate** [comment]
Hi,
Apologies for the late response. I am certainly happy to consider any pull request that allows liftoff to handle a wider ranger of GFF3s.
-Alaina

**Comment by Juke34** [comment]
This can be dangerous, `Derives_from` do not have the same meaning as "Parent". We should inform mirbase.org to be more compliant with the official specification: https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md 
Personnaly I'm not sure this is something that has to be done within LiftOff. Users should provide proper input files (And fix them if needed).

In this pipeline https://github.com/guigolab/FA-nf they added AGAT as GFF/GTF sanitizer, LiftOffit is not a pipeline, but we could provide a guideline how to sanitize GFF/GTF files before to use LiftOff.

**Comment by tdido** [comment]
Hi, thanks for your reply.

This is of course debatable, but I don't necessarily see it as dangerous, since the default behaviour is still using Parent. If the user is using arguments one could assume she/he knows what she/he is doing.

In particular, I guess miRBase considers the mature miRNA transcripts to be somewhat akin to Polycistronic transcripts as described in the document you linked, since there are different miRNA hairpins that generate the exact same mature sequence.

In any case, thank you for your feedback and feel free to close the request if you think it's not worth including.

**Comment by agshumate** [comment]
Hi,
I do agree that it is not dangerous since the default is still Parent so changing it from that is sort of 'at your own risk'. I also really like AGAT as a way to sanitize GFF/GTFs. I will add that to the readme and merge in the pull request unless there are further concerns about that.

**Comment by Juke34** [comment]
Right, `dangerous` was a bit strong ^^. I agree novice will not play with that parameter. 
I mainly wanted to warn that trying to be compliant with all sort of GFF/GTF types that can be met might be a never ending work.

**Comment by jolbi** [comment]
Hi,

Has this feature been implemented? I have miRNA gff like that. How can I tell liftoff to use  "Derives_from" field?
